packages/visualizations/src/components/Table/Cell/Format/ImageFormat.svelte

```diff
@@ -0,0 +1,79 @@
+<script lang="ts">
+    import type { Instance } from 'tippy.js';
+
+    import tippy from '../../../utils/tippy';
+    import { warn } from './utils';
+
+    export let rawValue: string;
+    export let display = (v: string) => v;
+    export let alt: (v: string) => string = () => '';
+    export let loadingContent: (v: string) => string = () => 'Loading...';
+    export let errorContent: (v: string) => string = () => 'Error while fetching image';
+
+    let tooltipContent: Element;
+    let image: HTMLImageElement | null = null;
+    let imageLoaded = false;
+    let hasError = false;
+
+    async function loadImage(tippyInstance: Instance) {
+        if (image !== null || hasError) return;
+        image = new Image();
+        image.src = rawValue;
+        try {
+            await image.decode();
+            image.alt = alt(rawValue);
+            image.classList.add('ods-visualization-table__image');
+            imageLoaded = true;
+        } catch (error) {
+            warn(rawValue, 'Image');
+            hasError = true;
+        }
+        tippyInstance.setContent(tooltipContent);
+    }
+</script>
+
+<div
+    role="button"
+    tabindex="0"
+    use:tippy={{
+        content: tooltipContent,
+        theme: 'ods-visualization-table',
+        delay: [500, 0],
+        duration: [275, 0],
+        maxWidth: 'none',
+        onTrigger(instance) {
+            loadImage(instance);
+        },
+    }}
+>
+    <div class="ods-visualization-table__image-label">{display(rawValue)}</div>
+    <div bind:this={tooltipContent}>
+        {#if imageLoaded}
+            {@html image?.outerHTML}
+        {:else if hasError}
+            <div class="ods-visualization-table__image-error-container">
+                {@html errorContent(rawValue)}
+            </div>
+        {:else}
+            <div class="ods-visualization-table__image-loading-container">
+                {@html loadingContent(rawValue)}
+            </div>
+        {/if}
```

Comment:
I don't think we should use `@html` to replace the lack of children in the SDK for now. It opens up a lot of security breaches (not for `new Image` which is safe) and forces the user to write HTML in template strings.
IMO, we should have something like this:

```suggestion
        <img 
            on:load={() => { isLoading = false; tippyInstance.setContent(...)}}
            on:error={(e) => { error = e; }}
            src={src}
            alt={alt}
        />
        {#if error}
            <div class="ods-visualization-table__image-error-container">
                {error}
            </div>
        {/if}
        {#if isLoading}
            <div class="ods-visualization-table__image-loading-container">
                maybe some default loader
            </div>
        {/if}
```
This leaves dom node creation to Svelte and limits errors to text and loader to pure CSS. These are limitations we should accept for now IMO.

Reply:
Users of this repository are developers. We are not responsible for any security breaches that may arise from the values displayed in the table. However, components APIs should be clear enough for developers to understand when they will directly change the DOM with an option.

We have a similar mechanism in place when displaying content in the MapWebGL popup (see: [GitHub link](https://github.com/opendatasoft/ods-dataviz-sdk/blob/2f2b66328c3de126055057a95dd42de816b4c220/packages/visualizations/src/components/Map/WebGl/WebGl.ts#L351)).


I still simplify the options to only allow customization of the loading and error messages.